megameklab/src/megameklab/ui/MegaMekLabTabbedUI.java

```diff
@@ -0,0 +1,352 @@
+
+        currentEditor().setEntity(entity, filename);
+        currentEditor().reloadTabs();
+        currentEditor().refreshAll();
+        // Set the tab name
+        tabs.setTabComponentAt(tabs.getSelectedIndex(), new ClosableTab(entity.getDisplayName(), currentEditor()));
+    }
+
+    @Override
+    public boolean exit() {
+        if (!currentEditor().safetyPrompt()) {
+            return false;
+        }
+
+        CConfig.setParam(CConfig.GUI_FULLSCREEN, Integer.toString(getExtendedState()));
+        CConfig.setParam(CConfig.GUI_PLAF, UIManager.getLookAndFeel().getClass().getName());
+        CConfig.writeMainUiWindowSettings(this);
+        CConfig.saveConfig();
+        PreferenceManager.getInstance().save();
+        MegaMek.getMMPreferences().saveToFile(MMLConstants.MM_PREFERENCES_FILE);
+        MegaMekLab.getMMLPreferences().saveToFile(MMLConstants.MML_PREFERENCES_FILE);
+
+        if (CConfig.getStartUpType() == MMLStartUp.RESTORE_TABS) {
+            try {
+                TabStateUtil.saveTabState(editors.stream().limit(editors.size() - 1).toList());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return true;
+    }
+
+    private void restrictToScrenSize() {
+        DisplayMode currentMonitor = getGraphicsConfiguration().getDevice().getDisplayMode();
+        int scaledMonitorW = UIUtil.getScaledScreenWidth(currentMonitor);
+        int scaledMonitorH = UIUtil.getScaledScreenHeight(currentMonitor);
+        int w = Math.min(getSize().width, scaledMonitorW);
+        int h = Math.min(getSize().height, scaledMonitorH);
+        setSize(new Dimension(w, h));
+    }
+
+
+    //<editor-fold desc="MenuBarOwner interface implementation">
+    @Override
+    public JFrame getFrame() {
+        return this;
+    }
+
+    @Override
+    public Entity getEntity() {
+        return currentEditor().getEntity();
+    }
+
+    @Override
+    public String getFileName() {
+        return currentEditor().getFileName();
+    }
+
+    @Override
+    public boolean hasEntityNameChanged() {
+        return currentEditor().hasEntityNameChanged();
+    }
+
+    @Override
+    public void refreshMenuBar() {
+        menuBar.refreshMenuBar();
+    }
+
+    @Override
+    public MenuBar getMMLMenuBar() {
+        return menuBar;
+    }
+    //</editor-fold>
+
+
+    /**
+     * Represents a button used for creating new tabs in the MegaMekLabTabbedUI interface.
+     * This class extends JPanel and is rendered as a non-opaque panel containing a "+" symbol.
+     * Used to mimic functionality for adding new tabs in a tabbed user interface.
+     * <p>
+     * The only reason this is a separate class instead of just the text for a tab is so that it can be green.
+     */
+    private static class NewTabButton extends JPanel {
+        public NewTabButton() {
+            setOpaque(false);
+            var label = new JLabel("➕");
```

Comment:
Request that you add:
```
            label.setFont(new Font("Symbola", Font.PLAIN, label.getFont().getSize()));
 ```
so that non-Windows users can see the symbols.